keeper-gradle-plugin/src/main/java/com/slack/keeper/KeeperPlugin.kt

```diff
@@ -106,62 +106,77 @@ public class KeeperPlugin : Plugin<Project> {
-      diagnosticOutputDir: Provider<Directory>
+      extension: KeeperExtension
   ) {
     afterEvaluate {
       if (extension.enableL8RuleSharing.getOrElse(false)) {
-        val r8Enabled = !hasProperty("android.enableR8") ||
-            property("android.enableR8")?.toString()?.toBoolean() != false
-        if (r8Enabled) {
-          appExtension.onApplicableVariants(project, extension) { testVariant, appVariant ->
-            val appR8Task = "minify${appVariant.name.capitalize(Locale.US)}WithR8"
-            val androidTestL8Task = "l8DexDesugarLib${testVariant.name.capitalize(Locale.US)}"
-            val inputFiles = tasks
-                .named<R8Task>(appR8Task)
-                .flatMap { it.projectOutputKeepRules }
-
-            tasks
-                .named<L8DexDesugarLibTask>(androidTestL8Task)
-                .configure {
-                  keepRulesFiles.from(inputFiles)
-                  keepRulesConfigurations.set(listOf("-dontobfuscate"))
-
-                  if (extension.emitDebugInformation.getOrElse(false)) {
-                    doFirst {
-                      val mergedFilesContent = keepRulesFiles.files.asSequence()
-                          .flatMap { it.walkTopDown() }
-                          .filterNot { it.isDirectory }
-                          .joinToString("\n") {
-                            "# Source: ${it.absolutePath}\n${it.readText()}"
+        appExtension.onApplicableVariants(project, extension) { testVariant, appVariant ->
+          val appR8Task = "minify${appVariant.name.capitalize(Locale.US)}WithR8"
+          val androidTestL8Task = "l8DexDesugarLib${testVariant.name.capitalize(Locale.US)}"
+          val inputFiles = tasks
+              .named<R8Task>(appR8Task)
+              .flatMap { it.projectOutputKeepRules }
+
+          tasks
+              .named<L8DexDesugarLibTask>(androidTestL8Task)
+              .configure {
+                val taskName = name
+                keepRulesFiles.from(inputFiles)
+                keepRulesConfigurations.set(listOf("-dontobfuscate"))
+                val diagnosticOutputDir = layout.buildDirectory.dir(
+                    "$INTERMEDIATES_DIR/l8-diagnostics/$taskName")
+                    .forUseAtConfigurationTime()
+                    .get()
+                    .asFile
+
+                // We can't actually declare this because AGP's NonIncrementalTask will clear it
+                // during the task action
+//                  outputs.dir(diagnosticOutputDir)
+//                      .withPropertyName("diagnosticsDir")
+
+                if (extension.emitDebugInformation.getOrElse(false)) {
+                  doFirst {
+                    val mergedFilesContent = keepRulesFiles.files.asSequence()
+                        .flatMap { it.walkTopDown() }
+                        .filterNot { it.isDirectory }
+                        .joinToString("\n") {
+                          "# Source: ${it.absolutePath}\n${it.readText()}"
```

Comment:
I don't know enough about what his is doing to know whether it's correct, but my inner linter is complaining about `absolutePath`.

Reply:
Yeah this is just a diagnostic and never used as a task input or anything